New platform support: GNU/Linux

snshn opened the pull request:
Compiles with zero warnings on Ubuntu with simple `make`.

Only two problems which are outside of the scope of this PR:
1. due to no keyboard control I cannot seem to rotate the model, trackpad driver here is not as good as on macOS, gestures are rudimentary.
2. no menu, since X11 just doesn't have that API built-in — it's a ton of code to add one, if needed at all.

P.S.
Wasn't trying to be a smart-ass in the README, my IDE trimmed trailing spaces automatically.

Commit:
add support for building on GNU/Linux

cubiq commented:
thanks for this! I'm able to compile but I'm getting a "Failed to initialize glfw" on run, any idea?

snshn commented:
@cubiq I found this for Ubuntu 18.04: https://github.com/glfw/glfw/issues/1814

If it's not Ubuntu, did you install both glfw and glfw-dev?  Can you run something basic, like `glxgears`?

cubiq commented:
thanks that helped. apparently you can't have both glfw for x11 and wayland. it's either/or. Once I fixed that for the current environment it worked.

snshn commented:
Oh, that's strange.  Glad it works now!

mkeeter commented:
Thanks!  Is this _actually_ using `gobjc++` (like it now says in the README)?  I don't see it being called anywhere, and the Makefile doesn't compile `platform/darwin.mm` in the Linux build.

snshn commented:
You are absolutely correct, I removed `gobjc++` and managed to successfully compile it on Ubuntu 20.04.  Don't remember where it came from, perhaps something I added by mistake while converting the macOS build process into GNU/Linux. 🤷🏻‍♂️

mkeeter commented:
Rebased + merged in 4e5a78fc87cde21f945579e6fbfae69b5f86734b, thanks!
